spec/ember-cli/build_constraint_spec.rb

```diff
@@ -0,0 +1,54 @@
+require "ember-cli/build_constraint"
+
+describe EmberCli::BuildConstraint do
+  describe "#enabled?" do
+    context "when #block is nil" do
+      it "returns true" do
+        constraint = EmberCli::BuildConstraint.new(block: nil, env: {})
+
+        expect(constraint.enabled?).to be true
+      end
+    end
+
+    context "when passed no arguments" do
+      it "captures the return value" do
+        invoked = false
+        constraint = EmberCli::BuildConstraint.new(
+          block: -> { invoked = true; :returned },
+          env: {},
+        )
+
+        expect(constraint.enabled?).to be :returned
+        expect(invoked).to be true
+      end
+    end
+
+    context "when passed a single argument" do
+      it "invokes the block with the path" do
+        path = "original"
+        constraint = EmberCli::BuildConstraint.new(
+          env: {},
+          block: -> p { path = "new-value"; :returned },
+        )
+
+        expect(constraint.enabled?).to be :returned
+        expect(path).to eq "new-value"
+      end
+    end
+
+    context "when passed two arguments" do
+      it "invokes the block with the path and Rack env" do
+        path = "old-path"
+        env = {}
+        constraint = EmberCli::BuildConstraint.new(
+          env: { "PATH_INFO" => "new-path" },
+          block: -> p, e { path = "new-path"; env = e; :returned },
```

Comment:
Unused block argument - `p`. If it's necessary, use `_` or `_p` as an argument name to indicate that it won't be used.<br>Do not use semicolons to terminate expressions.